src/utils.js

```diff
@@ -1,9 +1,24 @@
 // @flow
 
+import type { Coordinates } from './types';
+
 export const createRange = (start: number, end: number): Array<number> =>
   [...Array(1 + end - start).keys()].map(
     (index: number): number => start + index,
   );
 
 export const getRandomInt = (min: number, max: number): number =>
   Math.floor(Math.random() * (max - min + 1)) + min;
+
+export const getCurrentGPSCoordinates = (): Promise<Coordinates> =>
```

Comment:
Make this return a LoadObject and it will be much simpler

Reply:
How can I return LoadObject instead of promise here? If I do it with BaseAPI pattern this will be always cached, but I want to check new gps location every time I call the function. If I won't cache this will always return LoadObject.loading();